pyhive/presto.py

```diff
@@ -190,6 +190,34 @@ def _decode_binary(self, rows):
                 for row in rows:
                     row[i] = base64.b64decode(row[i])
 
+
+    def _pretty(self, columns, data):
+        r = []
+        for row in data: # the top-level is an iterable of records (i.e. rows)
+            c = []
+            for (column, row) in zip(columns, row):
+                c.append(self._pretty_(column["typeSignature"], row))
+            r.append(c)
+        return r
+
+
+    def _pretty_(self, column, data):
```

Comment:
If I'm reading this code correctly, the only edit it does is to attach column names to struct type fields. That's an incompatible change from a list to a dict, so could you do the transformations on your end of the code instead? (e.g. via a wrapper or subclass)